pymeasure/experiment/parameters.py

```diff
@@ -581,9 +658,79 @@ def eval_fget(self, parent):
             return fget
 
     def __str__(self):
-        result = self.fmt % self.value
+        # Deprecated, can be removed once the units and fmt parameters are removed from the
+        # Metadata base class
+        if hasattr(self, 'fmt'):
+            result = self.fmt % self.value
+        else:
+            result = super().__str__()
 
-        if self.units is not None:
+        if hasattr(self, 'units') and self.units is not None:
             result += " %s" % self.units
 
         return result
+
+
+class IntegerMetadata(Metadata, IntegerInputField):
+    pass
```

Comment:
Why do you have lengthy docstrings for the parameters, but not for the Metadata?

Naive, I would not add docstrings to either one.

Reply:
I simply did not get around to doing that. I was not yet completely finished with this, but mainly wondered about the structure with all the classes.